### ISSUE TITLE
Can not set posix permissions if there were no "WRITE" access at creation

### ISSUE THREAD

**Issue by glhez** [marschall/memoryfilesystem#50]
Hello,

Using the following code:

```java
  @Test
  public void test() throws IOException {
    try (FileSystem fs = MemoryFileSystemBuilder.newEmpty()
        .addFileAttributeView(DosFileAttributeView.class)
        .addFileAttributeView(PosixFileAttributeView.class)
        .build("perm")) {

      final Path path = fs.getPath("readable-at-first");
      Files.createFile(path,  PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("r--r--r--")));

      assertThat(Files.isReadable(path)).isTrue(); // ok
      assertThat(Files.isWritable(path)).isFalse(); // ok
      assertThat(Files.isExecutable(path)).isFalse(); // ok

      Files.setPosixFilePermissions(path, PosixFilePermissions.fromString("rw-r--r--")); // FAIL.

      assertThat(Files.isReadable(path)).isTrue(); // ok
      assertThat(Files.isWritable(path)).isTrue(); // (should be) ok
      assertThat(Files.isExecutable(path)).isFalse(); // ok
    }
  }
```

I get this exception:

```
java.nio.file.AccessDeniedException
	at com.github.marschall.memoryfilesystem.MemoryEntry$MemoryPosixFileAttributeView.checkAccess(MemoryEntry.java:810)
	at com.github.marschall.memoryfilesystem.MemoryEntry.checkAccess(MemoryEntry.java:168)
	at com.github.marschall.memoryfilesystem.MemoryEntry$MemoryPosixFileAttributeView.setPermissions(MemoryEntry.java:788)
```

This fails here:

```java
    @Override
    public void setPermissions(Set<PosixFilePermission> perms) throws IOException {
      if (perms == null) {
        throw new IllegalArgumentException("permissions must not be null");
      }
      try (AutoRelease lock = this.entry.writeLock()) {
        this.entry.checkAccess(AccessMode.WRITE); // <--- HERE.
        this.permissions = toMask(perms);
      }
    }
```

I think the check should not verify for the current permissions, but simply if the current owner is the file's owner, like if I were calling `chmod`.

On the other hand: delete the file works.

**Comment by marschall** [comment]
Gee, I had no idea.

**Comment by marschall** [comment]
Has been fixed in 0.6.7 which has been released to Maven Central. Sync may take a few hours.
